docs/shared/recipes/plugins/project-graph-plugins.md

```diff
@@ -80,6 +80,57 @@ If you create targets for a project within a plugin's code, the Nx migration gen
 
 {% /callout %}
 
+#### Example (extending projects / adding targets)
+
+A plugin can also be written to modify existing projects. Most of Nx's first party are written to add a target to a given project based on the configuration files present for that project. The below example shows how a plugin could add a target to a project based on the presence of a `tsconfig.json` file.
+
+```typescript {% fileName="/my-plugin/index.ts" %}
+export const createNodes: CreateNodes = [
+  '**/tsconfig.json',
+  (fileName: string, opts, context: CreateNodesContext) => {
+    const root = dirname(fileName);
+
+    const isProject = existsSync(join(root, 'project.json'));
+    if (!isProject) {
+      return {};
+    }
+
+    return {
+      projects: {
+        [root]: {
+          targets: {
+            build: {
+              command: `tsc -p ${fileName}`,
+            },
+          },
+        },
+      },
+    };
+  },
+];
+```
+
+By checking for the presence of a `project.json` file, the plugin can ensure that the project it is modifying is an existing Nx project.
```

Comment:
I think this is giving people a footgun.  This will fail for any projects that are identified by a `package.json`.  I don't think updating it to also check for a `package.json` would be enough either.  (Although maybe that's what we'll have to do.)

In order of preference, this is what I think the example code should do:
1. Infer different tasks based on whether the file is in an existing project (regardless of how that project was created)
2. Hard-code a check based on the two current most common ways of defining projects: `project.json` and `package.json`

I understand that this code is being run while the project graph is being created, so maybe we don't know what is and is not in a project definitively at this point.